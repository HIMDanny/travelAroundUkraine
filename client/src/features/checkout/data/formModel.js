export default {
  firstName: {
    name: 'firstName',
    label: 'First Name',
    requiredErrorMsg: 'First name is required',
  },

  lastName: {
    name: 'lastName',
    label: 'Last Name',
    requiredErrorMsg: 'Last name is required',
  },
  email: {
    name: 'email',
    label: 'Email',
    requiredErrorMsg: 'Email is required',
    invalidErrorMsg: 'Please enter a valid email',
  },
  phone: {
    name: 'phone',
    label: 'Phone',
    requiredErrorMsg: 'Phone is required',
  },
  addressOne: {
    name: 'addressOne',
    label: 'Address line 1',
    requiredErrorMsg: 'Address is required',
  },
  addressTwo: {
    name: 'addressTwo',
    label: 'Address line 2',
  },
  city: {
    name: 'city',
    label: 'City',
    requiredErrorMsg: 'City is required',
  },
  state: {
    name: 'state',
    label: 'State/Province/Region',
  },
  zipcode: {
    name: 'zipcode',
    label: 'Zipcode',
    requiredErrorMsg: 'Zipcode is required',
  },
  country: {
    name: 'country',
    label: 'Country',
    requiredErrorMsg: 'Country is required',
  },
  cardType: {
    name: 'cardType',
    label: 'Card Type',
    requiredErrorMsg: 'Card Type is required',
  },
  nameOnCard: {
    name: 'nameOnCard',
    label: 'Name on Card',
    requiredErrorMsg: 'Name on card is required',
  },
  cardNumber: {
    name: 'cardNumber',
    label: 'Card Number',
    requiredErrorMsg: 'Card number is required',
    invalidErrorMsg: 'Please enter a valid card number ex: 1234123412341234',
  },
  expiryDate: {
    name: 'expiryDate',
    label: 'Expiry Date',
    requiredErrorMsg: 'Expiry date is required',
  },
  cvv: {
    name: 'cvv',
    label: 'CVV',
    requiredErrorMsg: 'CVV is required',
  },
};
