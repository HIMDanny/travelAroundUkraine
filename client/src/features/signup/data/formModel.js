export default {
  firstName: {
    name: 'firstName',
    label: 'First Name',
    requiredErrorMsg: 'First name is required',
  },

  lastName: {
    name: 'lastName',
    label: 'Last Name',
    requiredErrorMsg: 'Last name is required',
  },

  login: {
    name: 'login',
    label: 'Login',
    requiredErrorMsg: 'Login is required',
  },

  email: {
    name: 'email',
    label: 'Email',
    requiredErrorMsg: 'Email is required',
    invalidErrorMsg: 'Please enter a valid email',
  },

  password: {
    name: 'password',
    label: 'Password',
    requiredErrorMsg: 'Password is required',
  },

  retypePassword: {
    name: 'retypePassword',
    label: 'Re-type Password',
    requiredErrorMsg: 'Please retype your password',
    invalidErrorMsg: 'Passwords must match',
  },

  telephone: {
    name: 'telephone',
    label: 'Phone',
  },

  birthdate: {
    name: 'birthdate',
    label: 'Birthdate',
  },

  gender: {
    name: 'gender',
    label: 'Gender',
  },

  avatarUrl: {
    name: 'avatarUrl',
    label: 'Please upload your image',
  },
};
