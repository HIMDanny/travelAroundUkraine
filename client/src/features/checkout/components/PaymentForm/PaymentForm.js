/* eslint-disable react/prop-types */
import React from 'react';
import { Grid, Typography, FormLabel, Radio, FormControlLabel, styled } from '@mui/material';
import { TextField, RadioGroup } from 'formik-mui';
import { DatePicker } from 'formik-mui-x-date-pickers';
import { Field } from 'formik';
import { formModel } from '../../data';

const PaymentForm = () => {
  const { cardType, nameOnCard, cardNumber, expiryDate, cvv } = formModel;

  const RadioGrouping = styled(RadioGroup)({
    justifyContent: 'space-evenly',
  });

  return (
    <>
      <Typography variant="h3" align="center" sx={{ mb: '20px' }}>
        Payment Details
      </Typography>

      <Grid container spacing={3} mb={3}>
        <Grid item xs={12} laptop={6}>
          <FormLabel id={cardType.name}>{cardType.label}</FormLabel>
          <Field component={RadioGrouping} aria-labelledby={cardType.name} row name={cardType.name}>
            <FormControlLabel
              componentsProps={{ typography: { gutterBottom: false } }}
              value="visa"
              control={<Radio />}
              label={<img src="./assets/images/checkoutPageForm/visa.png" alt="visa" />}
            />
            <FormControlLabel
              componentsProps={{ typography: { gutterBottom: false } }}
              value="mastercard"
              control={<Radio />}
              label={<img src="./assets/images/checkoutPageForm/mastercard.png" alt="mastercard" />}
            />
            <FormControlLabel
              componentsProps={{ typography: { gutterBottom: false } }}
              value="americanExpress"
              control={<Radio />}
              label={<img src="./assets/images/checkoutPageForm/americanExpress.png" alt="americanExpress" />}
            />
            <FormControlLabel
              componentsProps={{ typography: { gutterBottom: false } }}
              value="paypal"
              control={<Radio />}
              label={<img src="./assets/images/checkoutPageForm/paypal.png" alt="paypal" />}
            />
          </Field>
        </Grid>

        <Grid item xs={12} laptop={6}>
          <Field component={TextField} required name={cardNumber.name} label={cardNumber.label} fullWidth />
        </Grid>

        <Grid item xs={12} laptop={6}>
          <Field component={TextField} required name={nameOnCard.name} label={nameOnCard.label} fullWidth />
        </Grid>

        <Grid item xs={12} laptop={3}>
          <Field component={DatePicker} disablePast required name={expiryDate.name} label={expiryDate.label} />
        </Grid>

        <Grid item xs={12} laptop={3}>
          <Field component={TextField} required name={cvv.name} label={cvv.label} fullWidth />
        </Grid>
      </Grid>
    </>
  );
};

export default PaymentForm;
