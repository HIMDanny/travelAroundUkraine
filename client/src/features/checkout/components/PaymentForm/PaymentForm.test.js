/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import { Formik } from 'formik';
import PaymentForm from './PaymentForm';
import formModel from '../../data';

jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('@mui/material/MenuItem', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Grid', () => ({ children }) => <div>{children}</div>);

const FormikWrapper = ({ children }) => <Formik>{children}</Formik>;

describe('PaymentForm snapshot test', () => {
  test('should PaymentForm match snapshot', () => {
    const { asFragment } = render(
      <FormikWrapper>
        <PaymentForm checkoutFormModel={formModel} />
      </FormikWrapper>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
