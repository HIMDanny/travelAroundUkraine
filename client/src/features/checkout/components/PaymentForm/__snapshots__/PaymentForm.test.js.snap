// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentForm snapshot test should PaymentForm match snapshot 1`] = `
<DocumentFragment>
  <p>
    Payment Details
  </p>
  <div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Card Type
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby=":r0:-label :r0:"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            name="cardType"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Card Type
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r1:"
          id=":r1:-label"
        >
          Name on Card
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            name="nameOnCard"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Name on Card *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r2:"
          id=":r2:-label"
        >
          Card Number
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r2:"
            name="cardNumber"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Card Number *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r3:"
          id=":r3:-label"
        >
          Expiry Date
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r3:"
            name="expiryDate"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Expiry Date *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r4:"
          id=":r4:-label"
        >
          CVV
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r4:"
            name="cvv"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                CVV *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
